debezium-server-redis/src/main/java/io/debezium/server/redis/RedisStreamChangeConsumer.java

```diff
@@ -226,4 +236,14 @@ else if (isMemoryOk.check()) {
         // Mark the whole batch as finished once the sub batches completed
         committer.markBatchFinished();
     }
-}
\ No newline at end of file
+
+    private long getObjectSize(SimpleEntry<String, Map<String, String>> record) throws IOException {
```

Comment:
IMHO this will be a bit slow. We have `io.debezium.util.ApproximateStructSizeCalculator`. Maybe something similar can be introduced?